### PR TITLE
feat(ddm): add widget to dashboard

### LINKED ISSUE #59827: Have a button in scratchpad widget that will save it in dashboards (take inspiration from Discover)

### PR DESCRIPTION
**Pull request by obostjancic**
- Closes #59827

image

### CODE REVIEW

File: static/app/views/ddm/contextMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+function handleAddQueryToDashboard(
+  {projects, environments, datetime, op, mri}: MetricsQuery,
+  organization: Organization,
+  router: InjectedRouter,
+  displayType?: MetricDisplayType
+) {
+  const {start, end, period} = datetime;
+  // TODO(ddm): make a util that does this
+  const field = op ? `${op}(${mri})` : mri;
```
Comment (ArthurKnaus) [inline static/app/views/ddm/contextMenu.tsx:64]: Out of curiosity: do we support querying MRIs without operator?
I already have a similar util for alerts, can move it to ddm.

**Review by ArthurKnaus — APPROVED** [review]
lgtm 👍

**Review by matejminar — COMMENTED** [review]
I'm facing few issues:

1. group by is not preserved (nor display)
2. granularity changes when you choose a dashboard
3. displayType error when attempting to save

https://github.com/getsentry/sentry/assets/9060071/95b7e873-87dd-4211-812a-156201bb86f8

File: static/app/views/ddm/contextMenu.tsx (hunk 1 above)
Comment (obostjancic) [reply static/app/views/ddm/contextMenu.tsx:64]: I don't think we do, maybe we should throw an error or use a default if that can even happen. And yep, move the util to ddm then and i will reuse it here

Comment (ArthurKnaus) [reply static/app/views/ddm/contextMenu.tsx:64]: ATM I handle it as if no MRI was selected -> the create action is disabled.

**Comment by obostjancic** [comment]
> 1. group by is not preserved (nor display)
- done
> 2. granularity changes when you choose a dashboard
- yes, it defaults to whatever date time params are defined in the dashboard.
> 3. displayType error when attempting to save
- done

File: static/app/views/ddm/contextMenu.tsx (hunk 1 above)
Comment (obostjancic) [reply static/app/views/ddm/contextMenu.tsx:64]: makes sense, no op no party :D

**Pull request merged**